### PR TITLE
test: fix tests failing only on node v20

### PR DESCRIPTION
**Pull request by KhafraDev**
Fixes or skips (as a last resort) tests that consistently fail on v20, none of the flaky ones.

edit: windows 20 passed and ubuntu got stuck on the node-fetch tests (happens occasionally)

### CODE REVIEW

File: test/autoselectfamily.js [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
 const dgram = require('dgram')
 const { Resolver } = require('dns')
 const dnsPacket = require('dns-packet')
 const { createServer } = require('http')
 const { Client, Agent, request } = require('..')
-const { nodeHasAutoSelectFamily } = require('../lib/core/util')
+const { nodeHasAutoSelectFamily, nodeMajor } = require('../lib/core/util')
+
+if (nodeMajor >= 20) {
+  skip('some tests are failing')
```
Comment (KhafraDev) [inline test/autoselectfamily.js:12]: cc @ShogunPanda these tests are failing on v20 & I had no idea how to fix them.

**Review by ronag — APPROVED** [review]

File: test/autoselectfamily.js (hunk 1 above)
Comment (KhafraDev) [reply test/autoselectfamily.js:12]: the tests still seem to fail:

<details>

```
TAP version 13
# Subtest: with autoSelectFamily enable the request succeeds when using request
    1..3
    ok 1 - should not error
    ok 2 - should be equivalent strictly
    ok 3 - should be equivalent strictly
ok 1 - with autoSelectFamily enable the request succeeds when using request # time=34.663ms

# Subtest: with autoSelectFamily enable the request succeeds when using a client
    1..3
    ok 1 - should not error
    ok 2 - should be equivalent strictly
    ok 3 - should be equivalent strictly
ok 2 - with autoSelectFamily enable the request succeeds when using a client # time=8.505ms

# Subtest: with autoSelectFamily disabled the request fails when using request
    1..1
    not ok 1 - Cannot read properties of null (reading 'code')
      ---
      stack: >
        undici/test/autoselectfamily.js:156:26

        RequestHandler.onHeaders (undici/lib/api/api-request.js:105:14)

        Request.onHeaders (undici/lib/core/request.js:233:27)

        Parser.onHeadersComplete (undici/lib/client.js:794:23)

        wasm_on_headers_complete (undici/lib/client.js:408:30)

        null.<anonymous> (wasm://wasm/00036ac6:1:1173)

        null.<anonymous> (wasm://wasm/00036ac6:1:4100)

        null.<anonymous> (wasm://wasm/00036ac6:1:28727)

        null.<anonymous> (wasm://wasm/00036ac6:1:5454)
      at:
        line: 156
        column: 26
        file: undici/test/autoselectfamily.js
      type: TypeError
      tapCaught: uncaughtException
      test: with autoSelectFamily disabled the request fails when using request
      source: "      }, (err, { statusCode, body }) => {\r

        \        t.strictSame(err.code, 'ECONNREFUSED')\r

        -------------------------^

        \      })\r

        \    })\n"
      ...

    # failed 1 test
not ok 3 - with autoSelectFamily disabled the request fails when using request # time=27.343ms

# Subtest: with autoSelectFamily disabled the request fails when using a client
    1..1
    not ok 1 - Cannot read properties of null (reading 'code')
      ---
      stack: >
        undici/test/autoselectfamily.js:183:26

        RequestHandler.onHeaders (undici/lib/api/api-request.js:105:14)

        Request.onHeaders (undici/lib/core/request.js:233:27)

        Parser.onHeadersComplete (undici/lib/client.js:794:23)

        wasm_on_headers_complete (undici/lib/client.js:408:30)

        null.<anonymous> (wasm://wasm/00036ac6:1:1173)

        null.<anonymous> (wasm://wasm/00036ac6:1:4100)

        null.<anonymous> (wasm://wasm/00036ac6:1:28727)

        null.<anonymous> (wasm://wasm/00036ac6:1:5454)
      at:
        line: 183
        column: 26
        file: undici/test/autoselectfamily.js
      type: TypeError
      tapCaught: uncaughtException
      test: with autoSelectFamily disabled the request fails when using a client
      source: "      }, (err, { statusCode, body }) => {\r

        \        t.strictSame(err.code, 'ECONNREFUSED')\r

        -------------------------^

        \      })\r

        \    })\n"
      ...

    # failed 1 test
not ok 4 - with autoSelectFamily disabled the request fails when using a client # time=17.694ms

1..4
# failed 2 of 4 tests
# time=122.721ms
```

</details>

**Comment by ShogunPanda** [comment]
@KhafraDev Once nodejs/node#47860, nodejs/node#48189 and nodejs/node#48258 land in Node.js 20, you might want to revert this.

**Comment by ShogunPanda** [comment]
@KhafraDev 20.3.0 has landed with all family autoselection fixes. Do you mind trying to revert this and see if all issues are gone?

**Pull request merged**